base/array.jl

```diff
@@ -413,8 +413,11 @@ oneunit(x::AbstractMatrix{T}) where {T} = _one(oneunit(T), x)
 
 ## Conversions ##
 
-# arises in similar(dest, Pair{Union{},Union{}}) where dest::Dict:
-convert(::Type{Vector{Union{}}}, a::Vector{Union{}}) = a
+convert(::Type{T}, a::T) where {T<:Array} = a
+convert(::Type{T}, a::AbstractArray) where {T<:Array} = T(a)
+
+convert(::Type{AbstractArray{T}}, a::AbstractArray) where {T} = AbstractArray{T}(a)
```

Comment:
Did you try removing this one, too?

Reply:
Ah, nevermind.  I was getting confused by `Type{<:AbstractArray}` and `Type{AbstractArray}`.  This looks great.  For anyone else getting into the same confusion, this dispatches to the `AbstractArray{T}(…)` constructor, which definitely exists and just `copyto!`s into a `similar` array.